python/fapolicy_analyzer/ui/state_manager.py

```diff
@@ -83,17 +88,82 @@ def __update_dirty_queue(self):
         Returns True is there are unapplied changes, False otherwise."""
         if self.bDirtyQ != self.is_dirty_queue():
             self.bDirtyQ = self.is_dirty_queue()
-            self.changeset_queue_updated()
+            self.ev_changeset_queue_updated()
         return self.bDirtyQ
 
+    # ##################### Queue conversion utilities #####################
     def get_path_action_list(self):
         # Iterate through the StateManagers Changeset list
         # Each changeset contains a dict with at least one Path/Action pair
+        # The rationale for the tuple format is the Gtk TreeView widget display
         return [t for e in self.listChangeset for t in e.get_path_action_map().items()]
 
-    def add_system_notification(
-        self, notification: str, notification_type: NotificationType
-    ):
+    def __path_action_list_2_queue(self, listPathAction):
+        """Converts a list of Path/Action tuples to populate changeset queue
+This is a utility function intended for unit testing and troubleshooting
+the StateManager and its interactions with interfacing objects. It is used to
+populate the internal queue structure with Changeset objects, however it does
+not exercise the full normal end to end data path which will normally apply and
+deploy these changesets.
+"""
+        for e in listPathAction:
+            cs = Changeset()
+            if e[1] == "Add":
+                cs.add_trust(e[0])
+            elif e[1] == "Del":
+                cs.del_trust(e[0])
+            else:
+                print("Error: Path/Action Pair: "
+                      "Unknown Action: {} {}".format(e[0], e[1]))
+                continue
+            self.add_changeset_q(cs)
+
+    def __path_action_list_to_dict(self):
+        """Convert Path/Action list of tuple pairs to dict for json xfer"""
+        dictPA = dict()
+        for t in self.get_path_action_list():
+            dictPA[t[0]] = t[1]
+        logging.debug("Path/Action Dict: {}".format(dictPA))
+        return dictPA
+
+    def __path_action_dict_to_list(self, dictPathAction):
+        """Convert Path/Action dict to list of tuple pairs for json xfer"""
+        listPaTuples = list()
+        for e in dictPathAction.keys():
+            listPaTuples.append((e, dictPathAction[e]))
+        logging.debug("PathAction Tuple List: {}".format(listPaTuples))
+        return listPaTuples
```

Comment:
Similar to the above method this can become a one liner with comprehensions:
```python
return [(path, action) for (path, action) in dictPathAction.items()]
```

Reply:
Some of these conversions will go away after we migrate the elements of the StateManager Q.